contrib/auto-render/splitAtDelimiters.js

```diff
@@ -27,76 +27,72 @@ const findEndOfMath = function(delimiter, text, startIndex) {
     return -1;
 };
 
-const splitAtDelimiters = function(startData, leftDelim, rightDelim, display) {
-    const finalData = [];
+const escapeRegex = function(string) {
+    return string.replace(/[-/\\^$*+?.()|[\]{}]/g, "\\$&");
+};
+
 
-    for (let i = 0; i < startData.length; i++) {
-        if (startData[i].type === "text") {
-            const text = startData[i].data;
+const splitAtDelimiters = function(text, delimiters) {
+    let regexLeft;
+    for (let i = 0; i < delimiters.length; i++) {
+        if (i === 0) {
+            regexLeft = "(";
+        } else {
+            regexLeft += "|";
+        }
+        regexLeft += escapeRegex(delimiters[i].left);
+    }
+    regexLeft = new RegExp(regexLeft + ")");
 
-            let lookingForLeft = true;
-            let currIndex = 0;
-            let nextIndex;
+    let lookingForLeft = true;
+    let index;
+    const data = [];
 
-            nextIndex = text.indexOf(leftDelim);
-            if (nextIndex !== -1) {
-                currIndex = nextIndex;
-                finalData.push({
+    while (true) {
+        if (lookingForLeft) {
+            index = text.search(regexLeft);
+            if (index === -1) {
+                break;
+            }
+            if (index > 0) {
+                data.push({
                     type: "text",
-                    data: text.slice(0, currIndex),
+                    data: text.slice(0, index),
                 });
-                lookingForLeft = false;
+                text = text.slice(index);
             }
-
-            while (true) {
-                if (lookingForLeft) {
-                    nextIndex = text.indexOf(leftDelim, currIndex);
-                    if (nextIndex === -1) {
-                        break;
-                    }
-
-                    finalData.push({
-                        type: "text",
-                        data: text.slice(currIndex, nextIndex),
-                    });
-
-                    currIndex = nextIndex;
-                } else {
-                    nextIndex = findEndOfMath(
-                        rightDelim,
-                        text,
-                        currIndex + leftDelim.length);
-                    if (nextIndex === -1) {
-                        break;
-                    }
-
-                    finalData.push({
-                        type: "math",
-                        data: text.slice(
-                            currIndex + leftDelim.length,
-                            nextIndex),
-                        rawData: text.slice(
-                            currIndex,
-                            nextIndex + rightDelim.length),
-                        display: display,
-                    });
-
-                    currIndex = nextIndex + rightDelim.length;
-                }
-
-                lookingForLeft = !lookingForLeft;
+        } else {
+            let i = 0;
+            while (!text.startsWith(delimiters[i].left)) { i++; }
```

Comment:
This could be:
```
const delim = delimiters.findIndex(delim => text.startsWith(delim.left));
```
It might be worth adding a comment here as well saying why this should always succeed.